### ISSUE TITLE
The output shape is different between onnx and keras

### ISSUE THREAD

**Issue by reymondzzzz** [onnx/keras-onnx#252]
I have small nested backbone in keras convnet. I want to convert to onnx.
backbone:

```
def _bottleneck(x, filters, activation, strides, block_id):
    padding = 'same' if strides == 1 else 'valid'
    ch_axis = 1 if K.image_data_format() == 'channels_first' else -1
    if strides > 1:
        x = KL.ZeroPadding2D(((0, 1), (0, 1)), data_format=K.image_data_format())(x)

    x = KL.Conv2D(filters // 2, (1, 1), padding='same', name=f'bottleneck_{block_id}_conv_0',
                  use_bias=False, data_format=K.image_data_format())(x)
    x = KL.BatchNormalization(axis=ch_axis, name=f'bottleneck_{block_id}_bnorm_0')(x)

    if activation == 'relu':
        x = KL.ReLU(name=f'bottleneck_{block_id}_relu_0')(x)
    elif activation == 'leaky':
        x = KL.LeakyReLU(name=f'bottleneck_{block_id}_leaky_0')(x)
    else:
        assert False

    x = KL.Conv2D(filters // 2, (3, 3), padding=padding, name=f'bottleneck_{block_id}_conv_1',
                  strides=strides, use_bias=False, data_format=K.image_data_format())(x)
    x = KL.BatchNormalization(axis=ch_axis, name=f'bottleneck_{block_id}_bnorm_1')(x)
    if activation == 'relu':
        x = KL.ReLU(name=f'bottleneck_{block_id}_relu_1')(x)
    elif activation == 'leaky':
        x = KL.LeakyReLU(name=f'bottleneck_{block_id}_leaky_1')(x)
    else:
        assert False

    x = KL.Conv2D(filters, (1, 1), padding='same', name=f'bottleneck_{block_id}_conv_2',
                  use_bias=False, data_format=K.image_data_format())(x)
    x = KL.BatchNormalization(axis=ch_axis, name=f'bottleneck_{block_id}_bnorm_2')(x)
    if activation == 'relu':
        x = KL.ReLU(name=f'bottleneck_{block_id}_relu_2')(x)
    elif activation == 'leaky':
        x = KL.LeakyReLU(name=f'bottleneck_{block_id}_leaky_2')(x)
    else:
        assert False

    return x

def convnet_7(input_shape: Tuple[Union[int, None], Union[int, None], int],
              activation: str) -> KM.Model:
    input = KL.Input(shape=input_shape, name='input_1')
    x = _bottleneck(input, filters=16, strides=1, activation=activation, block_id=1)
    x = _bottleneck(x, filters=32, strides=2, activation=activation, block_id=2)
    x = _bottleneck(x, filters=64, strides=1, activation=activation, block_id=3)
    x = _bottleneck(x, filters=64, strides=1, activation=activation, block_id=4)
    x = _bottleneck(x, filters=128, strides=2, activation=activation, block_id=5)
    x = _bottleneck(x, filters=128, strides=1, activation=activation, block_id=6)
    x = _bottleneck(x, filters=128, strides=1, activation=activation, block_id=7)

    return KM.Model(inputs=input, outputs=x, name='convnet_7')
```
Unfortunately, i can not get head and it doesn’t mattter

If I just save model in onnx and run in onnxruntime i get

```
model = create_keras_model()
model.load_weights("blah_blah")
model = keras2onnx.convert_keras(model, name=model.name, doc_string='', target_opset=7)
onnx.save(model, "blah_1")

input = np.ones((1, 3, 96, 128)).astype(np.float32)

sess = rt.InferenceSession("blah_1.onnx")
input_name = sess.get_inputs()[0].name

pred_onx = sess.run(None, {input_name: input})

GENERAL ERROR : Load model from blah_1.onnx failed:Node:Identity1 Output:xy_conv_1/BiasAdd:0 [ShapeInferenceError] Can't merge shape info. Both source and target dimension have values but they differ. Source=23 Target=24 Dimension=2
```
And if I change in function _bottleneck all same paddings to valid, create new model and load old weigths, saving and run onnx model. it's worked correctly. 

Empty reproducing [model](https://github.com/onnx/keras-onnx/files/3663425/blah_1.onnx.zip)

Thank you!!!

**Comment by jiafatom** [comment]
Can you share the keras model? I want to use your script, but seems that the script you shared has a mismatch with your input shape `(1, 3, 96, 128)`. Then the keras model itself has issues.

**Comment by jiafatom** [comment]
I made a fix on [onnxconverter-common](https://github.com/microsoft/onnxconverter-common), you need manually build the package from onnxconverter-common master, and then test it. 
keras2onnx 1.5.2 release will be published very soon. You still need build onnxconverter-common master, though.

**Comment by reymondzzzz** [comment]
It's worked! Thanks. When next release onnxconverter-common? its need my team

**Comment by jiafatom** [comment]
Can your code work with onnxconverter-common 1.5.4?

**Comment by reymondzzzz** [comment]
@jiafatom 
```
import os
os.environ['TF_KERAS'] = '1'

import tensorflow.keras.models as KM
import tensorflow.keras.backend as K
import tensorflow.keras.layers as KL

import keras2onnx

import onnxruntime as rt
import onnx
import numpy as np

K.set_image_data_format("channels_first")

def _bottleneck(x, filters, activation, strides, block_id):
    padding = 'same' if strides == 1 else 'valid'
    ch_axis = 1 if K.image_data_format() == 'channels_first' else -1
    if strides > 1:
        x = KL.ZeroPadding2D(((0, 1), (0, 1)), data_format=K.image_data_format())(x)

    x = KL.Conv2D(filters // 2, (1, 1), padding='same', name=f'bottleneck_{block_id}_conv_0',
                  use_bias=False, data_format=K.image_data_format())(x)
    x = KL.BatchNormalization(axis=ch_axis, name=f'bottleneck_{block_id}_bnorm_0')(x)

    if activation == 'relu':
        x = KL.ReLU(name=f'bottleneck_{block_id}_relu_0')(x)
    elif activation == 'leaky':
        x = KL.LeakyReLU(name=f'bottleneck_{block_id}_leaky_0')(x)
    else:
        assert False

    x = KL.Conv2D(filters // 2, (3, 3), padding=padding, name=f'bottleneck_{block_id}_conv_1',
                  strides=strides, use_bias=False, data_format=K.image_data_format())(x)
    x = KL.BatchNormalization(axis=ch_axis, name=f'bottleneck_{block_id}_bnorm_1')(x)
    if activation == 'relu':
        x = KL.ReLU(name=f'bottleneck_{block_id}_relu_1')(x)
    elif activation == 'leaky':
        x = KL.LeakyReLU(name=f'bottleneck_{block_id}_leaky_1')(x)
    else:
        assert False

    x = KL.Conv2D(filters, (1, 1), padding='same', name=f'bottleneck_{block_id}_conv_2',
                  use_bias=False, data_format=K.image_data_format())(x)
    x = KL.BatchNormalization(axis=ch_axis, name=f'bottleneck_{block_id}_bnorm_2')(x)
    if activation == 'relu':
        x = KL.ReLU(name=f'bottleneck_{block_id}_relu_2')(x)
    elif activation == 'leaky':
        x = KL.LeakyReLU(name=f'bottleneck_{block_id}_leaky_2')(x)
    else:
        assert False

    return x

def convnet_7(input_shape,
              activation: str) -> KM.Model:
    input = KL.Input(shape=input_shape, name='input_1')
    x = _bottleneck(input, filters=16, strides=1, activation=activation, block_id=1)
    x = _bottleneck(x, filters=32, strides=2, activation=activation, block_id=2)
    x = _bottleneck(x, filters=64, strides=1, activation=activation, block_id=3)
    x = _bottleneck(x, filters=64, strides=1, activation=activation, block_id=4)
    x = _bottleneck(x, filters=128, strides=2, activation=activation, block_id=5)
    x = _bottleneck(x, filters=128, strides=1, activation=activation, block_id=6)
    x = _bottleneck(x, filters=128, strides=1, activation=activation, block_id=7)

    return KM.Model(inputs=input, outputs=x, name='convnet_7')


def _create_head(i, output_features_count: int, head_name: str):
    ch_axis = 1 if K.image_data_format() == 'channels_first' else -1

    x = KL.Conv2D(128, (3, 3), padding='same', name=f'{head_name}_conv_0',
                  use_bias=False, data_format=K.image_data_format())(i)

    x = KL.BatchNormalization(axis=ch_axis, name=f'{head_name}_bnorm_0')(x)
    x = KL.ReLU(name=f'{head_name}_relu_0')(x)

    x = KL.Conv2D(output_features_count, (1, 1), padding='same', name=f'{head_name}_conv_1',
                  use_bias=True, data_format=K.image_data_format())(x)

    return x

if __name__ == "__main__":
    img_input_shape = (3, 96, 96)
    model = convnet_7(input_shape=img_input_shape, activation="relu")
    i = KL.Input(img_input_shape, name='input')
    x = model(i)
    head = _create_head(x, output_features_count=1, head_name='head')
    base_model = KM.Model(inputs=[i], outputs=[head])
    base_model.summary()
    model_onnx = keras2onnx.convert_keras(base_model, name=base_model.name, doc_string='', target_opset=7)

    onnx.save(model_onnx, "model.onnx")
    input = np.ones((1, 3, 96, 96)).astype(np.float32)

    sess1 = rt.InferenceSession("model.onnx")
    input_name = sess1.get_inputs()[0].name

    pred_onx_1 = sess1.run(None, {input_name: input})

```

**Comment by reymondzzzz** [comment]
This code works on onnxconverter-common 1.5.4 from master.

**Comment by jiafatom** [comment]
Can it work when you `pip install onnxconverter-common` 1.5.4 version, not from master?

**Comment by reymondzzzz** [comment]
From pip is not worked. From master works

**Comment by jiafatom** [comment]
We just released onnxconverter-common 1.5.5 in pypi which includes this change, please update, thanks.
